Idrall/Resolve.idr

```diff
@@ -3,75 +3,137 @@ module Idrall.Resolve
 import Idrall.Error
 import Idrall.Expr
 import Idrall.IOEither
+import Idrall.Parser
+import Idrall.Path
+
+liftEither : Either e a -> IOEither e a
+liftEither = MkIOEither . pure
+
+mapErr : (e -> e') -> IOEither e a -> IOEither e' a
+mapErr f (MkIOEither x) = MkIOEither (do
+  x' <- x
+  (case x' of
+        (Left l) => pure (Left (f l))
+        (Right r) => pure (Right r)))
+
+parseErrorHandler : String -> Error
+parseErrorHandler x = ErrorMessage (x)
+
+fileErrorHandler : String -> FileError -> Error
+fileErrorHandler x y = ReadFileError (show y ++ " " ++ x)
+
+readFile' : String -> IOEither Error String
+readFile' x =
+  let c = MkIOEither (readFile x)
+      contents = mapErr (fileErrorHandler x) c in
+  contents
```

Comment:
```suggestion
  let c = MkIOEither (readFile x) in
    contents = mapErr (fileErrorHandler x) c
```

Reply:
think that's a syntax error, but i'll tidy it up in a similar way